core/src/main/java/google/registry/tools/server/RefreshDnsForAllDomainsAction.java

```diff
@@ -76,27 +80,39 @@ public class RefreshDnsForAllDomainsAction implements Runnable {
   public void run() {
     assertTldsExist(tlds);
     checkArgument(smearMinutes > 0, "Must specify a positive number of smear minutes");
-    tm().transact(
-            () ->
-                tm().query(
-                        "SELECT domainName FROM Domain "
-                            + "WHERE tld IN (:tlds) "
-                            + "AND deletionTime > :now",
-                        String.class)
-                    .setParameter("tlds", tlds)
-                    .setParameter("now", clock.nowUtc())
-                    .getResultStream()
-                    .forEach(
-                        domainName -> {
-                          try {
-                            // Smear the task execution time over the next N minutes.
-                            requestDomainDnsRefresh(
-                                domainName, Duration.standardMinutes(random.nextInt(smearMinutes)));
-                          } catch (Throwable t) {
-                            logger.atSevere().withCause(t).log(
-                                "Error while enqueuing DNS refresh for domain '%s'.", domainName);
-                            response.setStatus(HttpStatus.SC_INTERNAL_SERVER_ERROR);
-                          }
-                        }));
+    String lastInPreviousBatch = "";
+    while (true) {
+      String previousLastName = lastInPreviousBatch;
+      ImmutableList<String> domainBatch =
+          ImmutableList.copyOf(
+              tm().transact(
+                      () ->
+                          tm().query(
+                                  "SELECT domainName FROM Domain WHERE tld IN (:tlds) AND"
+                                      + " deletionTime > :now  AND domainName >"
```

Comment:
Do we need to refresh DNS on domains that might be in pending delete?  Wouldn't their DNS already be yanked?  It's possible this condition could simply be deletionTime = END OF TIME.